rott/rt_util.c

```diff
@@ -296,85 +297,94 @@ void ClearBuffer( char * buf, int size )
 /*
 =================
 =
-= Error
+= FileNotFoundError
 =
-= For abnormal program terminations
+= For when a required file is not found
 =
 =================
 */
 
-void Error (char *error, ...)
+void FileNotFoundError(const char *name)
 {
-   char msgbuf[300];
-	va_list	argptr;
-   int size;
-   char * sptr;
-   int level;
-   static int inerror = 0;
+	static char buffer[1024];
+	int len;
+	char *ptr;
+	const char **datadirs;
+	int num_datadirs;
+
+	datadirs = GetDataDirs(&num_datadirs);
 
+	SDL_snprintf(buffer, sizeof(buffer), "Required file not found: %s\n\nSearch paths:", name);
+	for (int i = 0; i < num_datadirs; i++)
+	{
+		// looks kinda weird to show in the list
+		if (datadirs[i][0] == '.')
+			continue;
 
-   inerror++;
-   if (inerror > 1)
-      abort();
+		len = SDL_strlen(buffer);
```

Comment:
`M_snprintf()` returns the length of the constructed string, so you just need to keep that instead of calculating the string length over and over again.

```c
int pos = 0;
char buf[];

pos += M_snprintf(buf + pos, ...);
pos += M_snprintf(buf + pos, ...);
...
```